crates/storage/store_db/libmdbx.rs

```diff
@@ -344,7 +385,7 @@ impl StoreEngine for Store {
             .map_err(StoreError::LibmdbxError)?;
         Ok(cursor
             .walk_key(transaction_hash.into(), None)
-            .map_while(|res| res.ok().map(|t| t.to()))
+            .map_while(|res| res.ok().map(|t| t.to().expect("RLP decode failed")))
```

Comment:
This should be handled differently or maybe it needs to be refactored, using `expect` has the same problems as using `unwrap`.

This said, try adding `expect_used = "deny"` to the clippy lints. If handling this implies a huge amount of work, file an issue.